### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,20 +1,18 @@
-const tags = require('./tags.js')
+const Tags = require('./tags.js')
+const attributes = ['language', 'source', 'encoding', 'resolution', 'dub'] // missing 'year', 'flags' (multi), 'type', 'group' and 'title'
```
Comment (soyuka) [inline src/index.js:2]: Can this be `Object.keys(Tags)`?

Comment (thcolin) [reply src/index.js:2]: Why not, but I don't know yet if `group`, `type` and `title` can be merged as one function (they will not be in `tags.js`) & rename to `properties`

File: src/index.js [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 
 class Release {
   constructor(name, strict = true, defaults = {}) {
-    defaults.language = (defaults.language || 'MULTi')
-    defaults.resolution = (defaults.resolution || 'SD')
+    defaults = Object.assign({language: 'MULTi', resolution: 'SD'}, defaults)
```
Comment (thcolin) [inline src/index.js:6]: 👍

File: src/index.js [hunk 3]
```diff
@@ -1,20 +1,18 @@
-const tags = require('./tags.js')
+const Tags = require('./tags.js')
```
Comment (thcolin) [inline src/index.js:1]: Rename to `Rules`, file too

File: src/index.js [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
-    var cleaned = this.clean(name)
-    var waste = cleaned
-    var attributes = ['language', 'source', 'encoding', 'resolution', 'dub'] // missing 'year', 'flags' (multi), 'type', 'group' and 'title'
+    const cleaned = this.clean(name)
+    let waste = cleaned
 
     this.original = name
     this.cleaned = cleaned
 
     attributes.map(key => {
```
Comment (thcolin) [inline src/index.js:14]: Rename `key` to `property`

File: src/index.js [hunk 5]
```diff
@@ -25,23 +23,22 @@ class Release {
...
   }
 
   parse(key, name) {
-    for (var i = 0; i < Object.keys(tags[key]).length; i++) {
-      var tag = Object.keys(tags[key])[i]
-      var patterns = (Array.isArray(tags[key][tag]) ? tags[key][tag] : [tags[key][tag]])
+    const tag = Tags[key]
+    const keys = Object.keys(tag)
+    for (let i = 0; i < keys.length; i++) {
+      const currentTag = keys[i]
```
Comment (thcolin) [inline src/index.js:33]: Rename to `tag`

File: src/index.js [hunk 6]
```diff
@@ -25,23 +23,22 @@ class Release {
...
-      .replace(/[\s]+/g, ' ')
-      .replace(/ /g, '.')
+    return name.replace(/[\[\]\(\)\;\:\!\s\\]+/g, '.')
   }
 
   parse(key, name) {
-    for (var i = 0; i < Object.keys(tags[key]).length; i++) {
-      var tag = Object.keys(tags[key])[i]
-      var patterns = (Array.isArray(tags[key][tag]) ? tags[key][tag] : [tags[key][tag]])
+    const tag = Tags[key]
```
Comment (thcolin) [inline src/index.js:30]: Rename to `rules`

File: src/index.js [hunk 7]
```diff
@@ -25,23 +23,22 @@ class Release {
...
+    const keys = Object.keys(tag)
```
Comment (thcolin) [inline src/index.js:31]: Rename to `tags`

**Review by thcolin — APPROVED** [review]
Approved, renaming are just a remainder for myself !

**Pull request merged**